### PR TITLE
Config for drones

### PR DESCRIPTION
**Pull request by ghost**
:cl: CosmicScientist
config: normal drone use can be disabled now, starts enabled
/:cl:

I wasn't sure where else to put the checks.

I assume and intend for spawning of drone mobs by admins to still be allowed even if the config isn't since it'd be under their supervision.

I intend for this to catch any situation where drones can be manufactured by players or placed on maps. ~~I don't know how to find and prevent construction/initialisation of their manufactories.~~

### CODE REVIEW

File: code/modules/mob/living/simple_animal/friendly/drone/drones_as_items.dm [hunk 1]
```diff
@@ -19,6 +19,8 @@
 
 /obj/item/drone_shell/Initialize()
 	. = ..()
+	if(!CONFIG_GET(flag/drone_ghost_role))
+		qdel(src)
```
Comment (kevinz000) [inline code/modules/mob/living/simple_animal/friendly/drone/drones_as_items.dm]: This is still terrible, if drones aren't enabled drone shell dispensers shouldn't accept metal/glass. This shouldn't be a silent metal/glass sink.

File: code/game/machinery/droneDispenser.dm [hunk 2]
```diff
@@ -144,6 +144,9 @@
 
 /obj/machinery/droneDispenser/process()
 	..()
+	if(!CONFIG_GET(flag/drone_ghost_role))
+		return
+
```
Comment (CosmicScientist) [inline code/game/machinery/droneDispenser.dm]: @kevinz000 this prevents production and I assume you can get the unspent materials out on attackby().

**Comment by Iamgoofball** [comment]
good fucking job lets move all balance methods to config now, just fire the design lead we dont need him its clear the headmins know how to do their job

**Comment by CosmicScientist** [comment]
>role is added to game
da rulez say you're not allowed to fix the cause of antag damage but you can fix the result
da drone page says the same
players can't seem to grasp this
players can't seem to grasp what living beings you're allowed to interact with (none)
admins can't agree on the finer details
rules lawyering
policy thread after policy thread

The players and admins can't agree on how you're supposed to Arr Pee the role, which it has been made dependent on given it's a ventcrawling dead crew replacement engineer. Nor is there a way to prevent wrongful interaction outside of assigning an admin to every drone.

If you go hunt down all the code and delete it, we won't need the config addition. Though I don't know if any downstreams use this role without issue and I doubted anyone wanted to get rid of drones before some magnificent coder, like you, could have fair warning to fix them.

**Comment by ExcessiveUseOfCobblestone** [comment]
this is not the proper solution though.

Remove it and downstreams can simply not merge it.

**Comment by KorPhaeron** [comment]
Drone rules are a dumpster fire and I'd like to see them reworked either in policy or in code but hiding them behind a config option is not really a fix

**Comment by CosmicScientist** [comment]
I'll see if I can come back with a removal PR then as it will fix unenforceable round interference limitation.

**Pull request closed without merging**